CMakeLists.txt

```diff
@@ -0,0 +1,286 @@
+cmake_minimum_required(VERSION 3.20)
+
+set(CMAKE_OSX_DEPLOYMENT_TARGET 10.15 CACHE STRING "Minimum macOS version targeted by sACNView on macOS")
+
+project(sACNView DESCRIPTION "A tool for sending and receiving the ANSI E1.31 Streaming ACN control protocol")
+
+# Only do these if this is the main project, and not if it is included through add_subdirectory
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+  ### Require out-of-source builds
+  file(TO_CMAKE_PATH "${PROJECT_BINARY_DIR}/CMakeLists.txt" LOC_PATH)
+  if(EXISTS "${LOC_PATH}")
+    message(FATAL_ERROR "You cannot build in a source directory (or any directory with a CMakeLists.txt file). Please make a build subdirectory. Feel free to remove CMakeCache.txt and CMakeFiles.")
+  endif()
+  
+  # Nicer IDE
+  set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+  # Workaround for CMake bug https://gitlab.kitware.com/cmake/cmake/-/issues/20812
+  if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    add_compile_options(
+      $<$<CONFIG:RELEASE>:/Zi>
+    )
+
+    add_link_options(
+      $<$<CONFIG:RELEASE>:/DEBUG>
+      $<$<CONFIG:RELEASE>:/OPT:REF>
+      $<$<CONFIG:RELEASE>:/OPT:ICF>
+    )
+  endif()
+  
+  # Docs only available if this is the main app
+  find_package(Doxygen)
+  if(Doxygen_FOUND)
+    add_subdirectory(docs)
```

Comment:
This doesn't work as although we have a docs directory, it isn't for Doxygen.

Reply:
Good point, I forgot to remove this section from my cmake template